### PR TITLE
[MINOR][DOCS] Fix documentation for slide function

### PR DESCRIPTION
**Pull request by bboutkov**
### What changes were proposed in this pull request?

This PR proposes to fix documentation for slide function. Fixed the spacing issue and added some parameter related info.

### Why are the changes needed?

Documentation improvement

### Does this PR introduce any user-facing change?

No (doc-only change).

### How was this patch tested?

Manually tested by documentation build.

### CODE REVIEW

**Comment by maropu** [comment]
Thanks for the first contribution, @bboutkov! Is this only one API without param docs there? Can you check the other APIs, too? cc: @HyukjinKwon

**Comment by maropu** [comment]
ok to test

**Comment by maropu** [comment]
retest this please

**Comment by bboutkov** [comment]
Hi. I didn't go through the rest of the docs, but looking now it does seem like there are a number other functions missing the params infos. 

I guess I wasn't sure how you all felt about adding them everywhere. Even for this function, the info is a bit redundant with the function summary so the need is not always clear to me, unless you have it as mandatory due to an internal policy. Honestly this issue more annoyed me due to the formatting bug :)

Maybe as a separate issue I can try and go through the rest of the docs and fix the param infos but unfortunately don't have time for that at the moment as it seems as its a pretty large number.

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -1973,6 +1973,11 @@ def slice(x, start, length):
     """
     Collection function: returns an array containing  all the elements in `x` from index `start`
     (array indices start at 1, or from the end if `start` is negative) with the specified `length`.
+
+    :param x: the array to be sliced
+    :param start: the starting index
+    :param length: the length of the slice
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py]: Okay, that's fine but can you at least update R and Scala APIs too to be consistent? You can add new params for start and length at https://github.com/apache/spark/blob/master/R/pkg/R/functions.R#L213 and add `@param`s at https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L3267-L3272

Comment (bboutkov) [reply python/pyspark/sql/functions.py]: Ok! will update shortly

Comment (bboutkov) [reply python/pyspark/sql/functions.py]: I added the scala params. 

Maybe I am mistaken, but it seems the R API already has this documented at https://github.com/apache/spark/blob/a9fbd310300e57ed58818d7347f3c3172701c491/R/pkg/R/functions.R#L3623

The rendering seems to work fine too on https://spark.apache.org/docs/latest/api/R/index.html so I am leaving the R docs alone.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -3267,6 +3267,11 @@ object functions {
   /**
    * Returns an array containing all the elements in `x` from index `start` (or starting from the
    * end if `start` is negative) with the specified `length`.
+   *
+   * @param x the column which is to be sliced
+   * @param start the starting index of the slice
+   * @param length the desired length of the slice
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: Why does it has the different statement from the one above? Can you make it consistent?

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 3]
```diff
@@ -3267,6 +3267,11 @@ object functions {
   /**
    * Returns an array containing all the elements in `x` from index `start` (or starting from the
    * end if `start` is negative) with the specified `length`.
+   *
+   * @param x the column which is to be sliced
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: nit: `column` -> `array column`?

**Review by maropu — APPROVED** [review]
Leave this to the final @HyukjinKwon check.

File: python/pyspark/sql/functions.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py]: Thanks. Can you match the wording while we're here?

**Review by HyukjinKwon — APPROVED** [review]
Thanks, @maropu. LGTM too except https://github.com/apache/spark/pull/26896/files#r358034648

File: python/pyspark/sql/functions.py (hunk 1 above)
Comment (bboutkov) [reply python/pyspark/sql/functions.py]: Fixed.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-2.4.

**Pull request closed without merging**